File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    // 인용 횟수 배열을 내림차순으로 정렬
    citations.sort((a, b) => b - a);
    // 배열을 순회하며 H-Index를 찾음
    for (let i = 0; i < citations.length; i++) {
        // 현재 인용 횟수
        let h = citations[i];
        
        // 조건을 만족하는지 확인
        if (h >= i + 1 && (i + 1 === citations.length || citations[i + 1] <= i + 1)) {
            return i + 1;
        }
    }
    return 0;
}

File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    citations.sort()
    ans = 0
    max_c = citations[-1] if citations else 0

    for h in range(max_c + 1):
        up = sum(c >= h for c in citations)   
        down = sum(c <= h for c in citations) 
        if up >= h and down <= h:
            ans = h
    return ans